go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1122,11 +1121,14 @@ func flowsForDefaultBridge(bridge *bridgeConfiguration, extraIPs []net.IP) ([]st
 			return nil, fmt.Errorf("unable to determine IPv4 physical IP of host: %v", err)
 		}
 		// table 0, SVC Hairpin from OVN destined to local host, DNAT and go to table 4
-		dftFlows = append(dftFlows,
-			fmt.Sprintf("cookie=%s, priority=500, in_port=%s, ip, ip_dst=%s, ip_src=%s,"+
-				"actions=ct(commit,zone=%d,nat(dst=%s),table=4)",
-				defaultOpenFlowCookie, ofPortPatch, config.Gateway.MasqueradeIPs.V4HostMasqueradeIP.String(), physicalIP.IP,
-				config.Default.HostMasqConntrackZone, physicalIP.IP))
+
+		for _, netConfig := range bridge.netConfig {
+			dftFlows = append(dftFlows,
+				fmt.Sprintf("cookie=%s, priority=500, in_port=%s, ip, ip_dst=%s, ip_src=%s,"+
+					"actions=ct(commit,zone=%d,nat(dst=%s),table=4)",
+					defaultOpenFlowCookie, netConfig.ofPortPatch, config.Gateway.MasqueradeIPs.V4HostMasqueradeIP.String(), physicalIP.IP,
+					config.Default.HostMasqConntrackZone, physicalIP.IP))
+		}
```

Comment:
@tssurya These are the default network openflow generation bits that I had to generalize for multiple networks support.  As discussed, maybe there should be extracted into dedicated functions that we call for the various netConfigs.

Reply:
on 2nd thoughts maybe its ok to keep them here, I tried to split them out as a new function but all I saw was heavy duplication with no benefits. specially given we won't be calling add network and adding netConfig's unless the featuregate for UDN is ON.

Reply:
I will go with what dumitru has done here and then if during reviews someone comes up with bright ideas we can fix it then

Reply:
I spoke to Tim, he thinks its ok, resolving this.